core/drivers/crypto/caam/hal/common/hal_rng.c

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   CAAM Random Number Generator Hardware Abstration Layer.
+ *         Implementation of primitives to access HW.
+ */
+#include <caam_hal_rng.h>
+#include <caam_io.h>
+#include <caam_status.h>
+#include <registers/rng_regs.h>
+#include <registers/version_regs.h>
+
+/*
+ * Returns if all RNG State Handler already instantiated or not
+ *
+ * @baseaddr  RNG Base Address
+ */
+bool caam_hal_rng_instantiated(vaddr_t baseaddr)
+{
+	uint32_t chaVid_ls = 0;
+	uint32_t nb_sh = 0;
+	uint32_t status = 0;
+
+	chaVid_ls = io_caam_read32(baseaddr + CHAVID_LS);
+
+	/* RNG version < 4 and RNG state handle is already instantiated */
+	if (GET_CHAVID_LS_RNGVID(chaVid_ls) < 4)
+		return true;
+
+	/* Get the Number of State Handles */
+	nb_sh = caam_hal_rng_get_nb_sh(baseaddr);
+
+	/* Read the RNG Status and checks if all channels are instantiatied */
+	status = caam_hal_rng_get_sh_status(baseaddr);
+
+	if (status != (uint32_t)((1 << nb_sh) - 1))
```

Comment:
I'd prefer
``` C
if (status != BIT32(nb_sh) - 1)
```
or maybe even depending on taste
``` C
if (status != GENMASK_32(nb_sh - 1, 0))
```
More compact and defined result for all values of `nb_sh`.

Reply:
ok